### PR TITLE
Release v0.4.1

### PR DESCRIPTION
**Pull request by kvid**
- [x] #346 
- [x] #347 
- [x] #358 
- [x] #361 
- [x] #363 
- [x] #364 
- [x] #371 
- [x] #374 
- [x] #375 
- [x] Cherry-pick doc addition https://github.com/wireviz/WireViz/commit/590504130c5e878c51df571974ee31e6f01b3a53 from `dev`
- [x] #380 
- [x] #381 
- [x] #383 
- [x] #392 
- [x] Print Python version & OS information when exception is raised in PR above
- [x] #395 
- [x] ~TODO: other fixes?~ Issue freeze
- [x] Interactive rebase to update PR refs
- [x] Update any missing `CHANGELOG` entries (remember to use new URL)
- [x] Apply `isort` & `black` on all `.py` files
-----
Should we avoid pushing these last steps to this PR branch? Or should they be pushed together just before PR merge into `master` and creating the release? Anything missing?
- [x] Bump version to v0.4.1
- [x] Execute `build_examples.py` and commit generated files
- [x] Update release date in CHANGELOG 
- [ ] Create release and publish it
-----
After the release:
- Apply all release changes to `dev` after or together with #251 merge-in to avoid regressions (manage in separate new PR!)

To try these fixes before release, this is one method:
```sh
pip uninstall wireviz
pip install git+https://github.com/wireviz/WireViz@release/v0.4.1-rc
```

### CODE REVIEW

**Comment by martinrieder** [comment]
I really appreciate your bugfixing work! Issue #257 is fixed, despite the discussion about strange corner cases.

**Comment by kvid** [comment]
@formatc1702 - I'm planning a rebase of this branch later today. Please tell me when you are finished merging in here for today, so I don't mess up your changes.

**Comment by formatc1702** [comment]
Thank you for all of this. I have merged some of the simpler bugs, feel free to rebase.
Also, you have full permission to directly merge/rebase other fixes into the 0.4.1rc branch yourself. 👍 
I will try to respond to the other new issues/PRs in the upcoming days.

**Comment by formatc1702** [comment]
Thanks. Let's not add more PRs unless they really are hotfixes. Thinks like #371 are already quite deep in feature-creep territory. This one is fine but everything else needs to wait.

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
+- Avoid Graphviz error when hiding all pins ([#257](https://github.com/wireviz/WireViz/issues/257), [#375](https://github.com/wireviz/WireViz/pull/375))
+- Avoid decimal point and trailing zero for integer BOM quantities ([#340](https://github.com/wireviz/WireViz/issues/340), [#374](https://github.com/wireviz/WireViz/pull/374))
+- Update project URL references ([#364](https://github.com/wireviz/WireViz/pull/364))
+- Add missing import of embed_svg_images ([#363](https://github.com/wireviz/WireViz/pull/363))
+- Use correct default title ([#360](https://github.com/wireviz/WireViz/issues/360), [#361](https://github.com/wireviz/WireViz/pull/361))
+- Fix bugs in mate processing ([#355](https://github.com/wireviz/WireViz/issues/355), [#358](https://github.com/wireviz/WireViz/pull/358))
+- Include missing files in published package ([#345](https://github.com/wireviz/WireViz/issues/345), [#347](https://github.com/wireviz/WireViz/pull/347)) 
+- Catch OSError(errno=EINVAL) ([#344](https://github.com/wireviz/WireViz/issues/344), [#346](https://github.com/wireviz/WireViz/pull/346))
```
Comment (formatc1702) [inline docs/CHANGELOG.md:22]: ❤️

**Comment by formatc1702** [comment]
Looks like this is pretty much ready.
Let's freeze this version by not adding any more bugfixes.

Is it necessary to do interactive rebase? For me, the integration of the bugfixes looks quite clean, and I'm trying to move away a bit from commit history perfectionism ;)
Otherwise, @kvid I suggest you let me know when it's done on your end by requesting a review, and I will take care of publishing.

The only note I have is: the to-do item "Bump version to v0.4.1-rc" should be "Bump version to v0.4.1" so that the generated examples bear the correct (released) version mark.

**Review by formatc1702 — APPROVED** [review]

**Comment by formatc1702** [comment]
> Should we avoid pushing these last steps to this PR branch? Or should they be pushed together just before PR merge into master and creating the release? Anything missing?

I've followed the same procedure as for previous releases. These steps are the last commits within the PR branch before merging into master. PyPI release will follow shortly.

**Comment by formatc1702** [comment]
WireViz v0.4.1 is released on PyPI 🚀 
https://pypi.org/project/wireviz/0.4.1/

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,31 @@
 # Change Log
 
-## [0.4](https://github.com/formatc1702/WireViz/tree/v0.4) (2024-05-12)
+## [0.4.1] (2024-07-13)
```
Comment (formatc1702) [inline docs/CHANGELOG.md:3]: Noticed too late that the link to the release is missing
```
## [0.4.1](https://github.com/formatc1702/WireViz/tree/v0.4) (2024-07-13)
```
but no biggie, can be fixed in next release.

Comment (kvid) [reply docs/CHANGELOG.md:3]: @formatc1702 wrote:
> Noticed too late that the link to the release is missing
> 
> ```
> ## [0.4.1](https://github.com/formatc1702/WireViz/tree/v0.4.1) (2024-07-13)
> ```

Remember not using the old URLs containing your personal user name `formatc1702`!

Another thing about the link: Now that you've created release pages for the version tags, should the version links in CHANGELOG point to the code tree for the tag (as today), or should they point to the release page for the tag?  https://github.com/wireviz/WireViz/releases/tag/v0.4.1

> 
> but no biggie, can be fixed in next release.

True. However, adding a minor commit to `master` fixing only such meta-information (not changing any file used by the code) long before creating a new release might also be considered, as far as I can understand, but I also can accept a point of view against that...

**Pull request merged**